src/test/java/com/shapesecurity/shift/serialization/Serializer.java

```diff
@@ -179,7 +181,13 @@ JsonObjectBuilder add(@NotNull String property, @NotNull Either<StringBuilder, S
       return this;
     }
 
-    StringBuilder done() {
+    StringBuilder done(@NotNull Node node) {
+      SourceLocation loc = node.getLoc();
+      if (loc != null) {
+        this.add("range",
```

Comment:
It is easy to do the sanity check. BTW, the end index is much more important than the source but we cannot have it without providing the source right now. I believe it's a defect.